contracts/src/v0.8/ccip/test/offRamp/OffRamp/OffRamp.trialExecute.t.sol

```diff
@@ -117,4 +122,68 @@ contract OffRamp_trialExecute is OffRampSetup {
     assertEq(uint256(Internal.MessageExecutionState.FAILURE), uint256(newState));
     assertEq(abi.encodeWithSelector(OffRamp.NotACompatiblePool.selector, address(0)), err);
   }
+
+  function test_trialExecute_CallWithExactGasRevertsAndSenderIsNotGasEstimator() public {
+    uint256[] memory amounts = new uint256[](2);
+    amounts[0] = 1000;
+    amounts[1] = 50;
+    Internal.Any2EVMRampMessage memory message =
+      _generateAny2EVMMessageWithTokens(SOURCE_CHAIN_SELECTOR_1, ON_RAMP_ADDRESS_1, 1, amounts);
+
+    bytes[] memory offchainTokenData = new bytes[](message.tokenAmounts.length);
+    uint32[] memory tokenGasOverrides = new uint32[](0);
+
+    vm.mockCallRevert(
+      address(s_offRamp),
+      abi.encodeWithSelector(s_offRamp.executeSingleMessage.selector, message, offchainTokenData, tokenGasOverrides),
+      abi.encodeWithSelector(CallWithExactGas.NOT_ENOUGH_GAS_FOR_CALL_SIG, "")
+    );
+
+    (Internal.MessageExecutionState newState, bytes memory err) =
+      s_offRamp.trialExecute(message, offchainTokenData, tokenGasOverrides);
+    assertEq(uint256(Internal.MessageExecutionState.FAILURE), uint256(newState));
+    assertEq(CallWithExactGas.NotEnoughGasForCall.selector, bytes4(err));
+  }
+
+  function test_trialExecute_RevertsWhen_NoGasForCallExactCheckAndSenderIsGasEstimator() public {
+    uint256[] memory amounts = new uint256[](2);
+    amounts[0] = 1000;
+    amounts[1] = 50;
+    Internal.Any2EVMRampMessage memory message =
+      _generateAny2EVMMessageWithTokens(SOURCE_CHAIN_SELECTOR_1, ON_RAMP_ADDRESS_1, 1, amounts);
+
+    bytes[] memory offchainTokenData = new bytes[](message.tokenAmounts.length);
+    uint32[] memory tokenGasOverrides = new uint32[](0);
+
+    vm.mockCallRevert(
+      address(s_offRamp),
+      abi.encodeWithSelector(s_offRamp.executeSingleMessage.selector, message, offchainTokenData, tokenGasOverrides),
+      abi.encodeWithSelector(CallWithExactGas.NO_GAS_FOR_CALL_EXACT_CHECK_SIG, "")
+    );
+
+    vm.expectRevert(MultiOCR3Base.InsufficientGasForCallWithExact.selector);
+    changePrank(GAS_ESTIMATION_SENDER);
```

Comment:
nit: no code should be between the expectRevert and the call, you can more the prank up